### PR TITLE
Smurf file emulator

### PR DESCRIPTION
**Pull request by jlashner**
Basic agent to generate fake files in the same directory structure that we are using for smurf G3 and auxiliary data.

## Description
This File-emulator agent is for the end-to-end DAQ tests, and generates fake files in the same directory structure that we use for the G3 and pysmurf data coming from the smurf-servers. 

There currently only exists a single process, `stream` that will write a fake IV file and then continuously generate fake g3 files at regular intervals. The fake G3 files currently just contain start/stop time, the number of channels and number of samples in a yaml readable format, but it's easy to build this out if we want more information or structure.. 

## Motivation and Context
End-to-end daq testing.

## How Has This Been Tested?
Tested running on smurf-srv-so3.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [ ] Unless I am preparing a release, I have opened this PR onto the `develop` branch.

### CODE REVIEW

**Review by BrianJKoopman — COMMENTED** [review]
Thanks Jack, this is a great starting point. Just to capture some of our discussion on the phone the other day -- we'd like to be able, for the end to end testing, to drop this Agent in place of the PysmurfControllerAgent so that the control program used to interface with the SMuRFs can make identical task/process calls to what will be in the Controller Agent. This will allow us to toggle some test flag on the control script end to emulate more closely the end to end process of collecting data. Once testing is completed the flag can be removed and the same control script functions can be used for controlling the real Agents.

Other small comments inline below.

File: agents/smurf_file_emulator/smurf_file_emulator.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+import os
+import argparse
+import warnings
+import txaio
+
+from typing import Optional
+
+on_rtd = os.environ.get('READTHEDOCS') == 'True'
+if not on_rtd:
+    from ocs import ocs_agent, site_config
```
Comment (BrianJKoopman) [inline agents/smurf_file_emulator/smurf_file_emulator.py:11]: With the newish mocking of packages in the readthedocs conf.py we no longer need this protection. And actually this protection breaks the readthedocs build. See: https://readthedocs.org/projects/socs/builds/15307869/

File: docs/agents/smurf_file_emulator.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
+.. highlight:: rst
+
+.. _smurf_file_emulator:
+
+====================
+Smurf File Emulator
+====================
+
+The Smurf File Emulator agent creates fake pysmurf and g3 files using the same
+directory structure that we're currently archiving on simons1. This is for
```
[the remark this answers is not in the document]
Comment (jlashner) [reply docs/agents/smurf_file_emulator.rst:10]: This is written the way files will be archived on the DAQ node, not as organized on the smurf-server

**Comment by jlashner** [comment]
Hi Brian, just implemented your suggestions and added more functions based on what I think will exist in the pysmurf controller.

**Review by BrianJKoopman — APPROVED** [review]
This is great, thanks Jack!

One thing I noticed when testing (which looks like it might have been true before recent changes too) is that the minimum length of a streamed file is the given file-duration, since the sleep before writing the file is that full length and can't be interrupted. My situation was that I set file-duration to 60 seconds in my config file, but then only turned the stream on for a few seconds. It took the full 60 seconds before a file was written, and it looked like it streamed for the full 60 seconds.

We might run into this in testing eventually. The delayed file output might be fine, but then we might have streams which overlap with other files, i.e. the bias step taken after we "finish" streaming starts during a stream. In a script like:
```
smurf_client.take_bias_steps()
acu_client.scan.start()
smurf_client.stream.start()
# time goes by while we scan
smurf_client.stream.stop()
acu_client.scan.stop()
smurf_client.take_bias_steps()
```

I'm not sure at the moment what effect this would have downstream during testing.

If you have time to fix that now, great. If not, I think this is a good starting point for testing and we could merge. That fix could be done later when we start using this if we see that it's a problem.

**Comment by jlashner** [comment]
Hey Brian, I improved the streamer loop to solve the problem you mentioned above and fixed another bug where incomplete files were not being written on stop. Let me know if there's anything else!

**Review by BrianJKoopman — APPROVED** [review]
Looks good! Thanks for the updates!

**Pull request merged**